x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotRequest.java

```diff
@@ -0,0 +1,435 @@
+     */
+    public MountSearchableSnapshotRequest ignoreIndexSettings(String... ignoreIndexSettings) {
+        this.ignoreIndexSettings = ignoreIndexSettings;
+        return this;
+    }
+
+    /**
+     * Sets the list of index settings and index settings groups that shouldn't be restored from snapshot
+     */
+    public MountSearchableSnapshotRequest ignoreIndexSettings(List<String> ignoreIndexSettings) {
+        this.ignoreIndexSettings = ignoreIndexSettings.toArray(new String[ignoreIndexSettings.size()]);
+        return this;
+    }
+
+    /**
+     * Returns the list of index settings and index settings groups that shouldn't be restored from snapshot
+     */
+    public String[] ignoreIndexSettings() {
+        return ignoreIndexSettings;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Settings settings) {
+        this.indexSettings = settings;
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Settings.Builder settings) {
+        this.indexSettings = settings.build();
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(String source, XContentType xContentType) {
+        this.indexSettings = Settings.builder().loadFromSource(source, xContentType).build();
+        return this;
+    }
+
+    /**
+     * Sets settings that should be added/changed for restored index
+     */
+    public MountSearchableSnapshotRequest indexSettings(Map<String, Object> source) {
+        try {
+            XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+            builder.map(source);
+            indexSettings(Strings.toString(builder), builder.contentType());
+        } catch (IOException e) {
+            throw new ElasticsearchGenerationException("Failed to generate [" + source + "]", e);
+        }
+        return this;
+    }
+
+    /**
+     * Returns settings that should be added/changed for restored index
+     */
+    public Settings indexSettings() {
+        return this.indexSettings;
+    }
+
+    /**
+     * Parses mount definition
+     *
+     * @param source mount definition
+     * @return this request
+     */
+    @SuppressWarnings("unchecked")
+    public MountSearchableSnapshotRequest source(Map<String, Object> source) {
+        // TODO: this could be rewritten with a Constructing Object Parser rather than manually
+        //  parsing the XContent, this was copied/modified from RestoreSnapshotRequest
+        for (Map.Entry<String, Object> entry : source.entrySet()) {
+            String name = entry.getKey();
+            if (name.equals("snapshot_index")) {
```

Comment:
Should we use  ̀indices` or `index`?

Reply:
`index` I think. We can't actually support restoring multiple indices currently, as there's no way to specify index-specific settings when restoring multiple indices (ie, restore `foo`, `bar`, and `baz`, but have different `index_settings` for each of them).